### PR TITLE
Feat: Add Support for DB Instance Role Associations

### LINKED ISSUE #131: Add support for RDS DB Instance role associations
## Describe the Feature

Allow the module to internally configure `aws_db_instance_role_association` resources, to centralize more of the RDS configuration in this single module. 

## Describe Ideal Solution

Example code snippet, using `for_each` to have the TF state being indexed by the feature name.

```hcl
variable "role_associations" {
  type        = map(string)
  description = "Map of role associations for the DB Instance. Key is the name of the feature for association, and value is the IAM Role ARN to associate. (e.g. { s3Import: rn:aws:iam::1234567890:role/rds-s3-import-role })"
  default     = {}
}

resource "aws_db_instance_role_association" "default" {
  for_each = module.this.enabled ? var.role_associations : {}

  db_instance_identifier = join("", aws_db_instance.default.*.id)
  feature_name           = each.key
  role_arn               = each.value
}
```

## Alternatives Considered

Current alternative is to add these role associations using a separate `resource` block.

### PR DESCRIPTION
**Pull request by korenyoni**
## what
* Add support for [DB Instance Role Associations](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/db_instance_role_association)
* Add BridgeCrew exceptions for errors stemming from this module parameterizing some resource attributes (false positives).
* Expand Terratest tests.
* Bump minimum supported Terraform version to `1.0.0` (see below).

## why
* Some RDS DB engines support different features depending on the engine. These should be supported by the module.
* The comprehensive Terratest suite in `examples/complete` and `test/src` will only work with Terraform `1.0.0` and above, and otherwise will result in "count cannot be determined before apply" errors.

## references
* Closes #131

### CODE REVIEW

File: examples/complete/main.tf
Comment (nitrocode) [inline examples/complete/main.tf:74]: can we keep locals at the top?

**Review by nitrocode — APPROVED** [review]

File: examples/complete/main.tf
Comment (korenyoni) [reply examples/complete/main.tf:74]: Keeping locals at the top is definitely the style we usually want, but in this case I created at a certain point near the bottom of the file because it is part of a section that is only applicable to the `testExamplesS3Integration` test case.

File: versions.tf
Comment (nitrocode) [inline versions.tf:2]: Woah. This is the first time we're doing this I believe.

Comment (nitrocode) [reply versions.tf:2]: We still have engagements using older versions. I think we'll need another approval on this if this is needed. Could you explain why this is being bumped to 1.0 ?

Comment (korenyoni) [reply versions.tf:2]: There's an issue with https://github.com/cloudposse/terraform-aws-s3-bucket/blob/master/main.tf#L376 not being able to be determined when the module enabled=false

But I see

Comment (korenyoni) [reply versions.tf:2]: I did explain it briefly in the _why_ section of my PR description

> The comprehensive Terratest suite in examples/complete and test/src will only work with Terraform 1.0.0 and above, and otherwise will result in "count cannot be determined before apply" errors.

**Comment by adamantike** [comment]
Bump! Is there anything blocking this PR from being merged, that we can help with?

File: examples/complete/main.tf
Comment (nitrocode) [inline examples/complete/main.tf:147]:
```suggestion
  version = "0.16.2"
```

File: examples/complete/main.tf
Comment (nitrocode) [inline examples/complete/main.tf:118]: Shouldn't we set these ?

**Review by nitrocode — CHANGES_REQUESTED** [review]
Please see comments

File: examples/complete/main.tf
Comment (nitrocode) [inline examples/complete/main.tf:124]: I think this may be causing the failure

**Pull request closed without merging**
